lib/pbench/agent/tool_meister.py

```diff
@@ -348,15 +340,13 @@ def wait(self):
         waits for the tool's start process to complete (since the "stop"
         process is supposed stop the "start" process).
         """
-        assert self.tool_dir is not None, "Logic bomb!  no tool directory provided!"
-        if not self.tool_dir.is_dir():
-            raise RuntimeError(f"tool directory does not exist: {self.tool_dir}")
-        if self.stop_process is None:
-            raise ToolException(f"Tool({self.name}) wait not called after 'stop'")
-        if self.start_process is None:
-            raise ToolException(
-                f"Tool({self.name}) does not have a start process running"
-            )
+        assert self.tool_dir is None, "Logic bomb!  tool directory still provided!"
```

Comment:
As I discussed elsewhere, it is a here because it is documenting the state that the Tool object should be in when `.wait()` is called.  That is, `.stop()` was called first to set it to `None`.

Reply:
The problem is, it's lousy documentation -- it would be much better if you replaced it with a code comment with the explanation that you just provided, that `.stop()` must be called prior to calling `.wait()`.

Reply:
Yeah; again, this is checking "internal state," sure; but internal state that's directly affected by the external caller. I don't like a fatal assert (which by convention an external caller *shouldn't* be catching) to tell the caller that it violated the class's sequence rules on method calls. I'd prefer an exception like `ToolMeisterBadSequence` informing the caller that `stop` needs to be called before `wait`, for example.

As long as Webb's unresolved ... 👀 🍤

Reply:
Apparently Peter is still declining, so I shall re-resolve.  🍤